scripts/report_incomplete_assignments_with_work_orders.py

```diff
@@ -0,0 +1,151 @@
+	"""
+	Setup logging
+	:param log_file: (string) The file to log to
+	:return: (Logger) a logging instance
+	"""
+	# initialize logging
+	formatter = logging.Formatter(
+		"[%(asctime)s] [%(filename)30s:%(lineno)4s - %(funcName)30s()][%(threadName)5s] [%(name)10.10s] [%(levelname)8s] %(message)s")
+	# Grab the root logger
+	logger = logging.getLogger()
+	# Set the root logger logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
+	logger.setLevel(logging.DEBUG)
+	# Create a handler to print to the console
+	sh = logging.StreamHandler(sys.stdout)
+	sh.setFormatter(formatter)
+	sh.setLevel(logging.INFO)
+	# Create a handler to log to the specified file
+	if log_file:
+		rh = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760)
+		rh.setFormatter(formatter)
+		rh.setLevel(logging.DEBUG)
+		logger.addHandler(rh)
+	# Add the handlers to the root logger
+	logger.addHandler(sh)
+	return logger
+
+
+def main(arguments):
+	# Initialize logging
+	logger = initialize_logging(arguments.log_file)
+
+	# Create the GIS
+	logger.info("Authenticating...")
+
+	# First step is to get authenticate and get a valid token
+	gis = GIS(arguments.org_url,
+			  username=arguments.username,
+			  password=arguments.password,
+			  verify_cert=not arguments.skip_ssl_verification)
+
+	logger.info("Getting workforce project")
+
+	# Get the workforce project
+	item = gis.content.get(arguments.project_id)
+	try:
+		project = workforce.Project(item)
+	except Exception:
+		logger.info("Invalid project id")
+		sys.exit(0)
+		
+	layer = None
+	# Get Survey or Collector Feature Layer
+	if arguments.survey_id and arguments.layer_url:
+		logger.info("Please try again with either survey id or layer url provided, not both")
+		sys.exit(0)
+	elif arguments.survey_id:
+		survey_item = gis.content.get(arguments.survey_id)
+		if survey_item:
+			layer = survey_item.layers[0]
+	elif arguments.layer_url:
+		layer = FeatureLayer(arguments.layer_url)
+	else:
+		logger.info("Please provide either a portal id for your survey feature layer or a feature service URL for your survey/collector layer")
+		sys.exit(0)
+		
+	# Check if layer exists
+	try:
+		json = layer._lyr_json
+	except Exception as e:
+		logger.info(e)
+		logger.info("Layer could not be found. Please check your parameters again. Exiting the script")
+		sys.exit(0)
+	
+	# Updating Assignments
+	logger.info("Querying assignments")
+	assignments = project.assignments.search()
+	for assignment in assignments:
+		if assignment.work_order_id and (assignment.status == "unassigned" or assignment.status == "assigned" or assignment.status == "declined"):
+			where = f"{arguments.field_name} = '{assignment.work_order_id}'"
+			if len(layer.query(where=where).features) > 0:
+				logger.info(f"Potential Assignment to Cancel: {str(assignment)} with OBJECTID {assignment.object_id}")
+				logger.info(f"Assignment Link: https://workforce.arcgis.com/projects/{arguments.project_id}/dispatch/assignments/{assignment.object_id}")
```

Comment:
Will need to be derived for Enterprise, You can use portal/self properties to build this I think.